Project-5.md

```diff
@@ -46,9 +46,501 @@ You may find the following resources useful:
+
+c.	~(a -> a)
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:IrreflexiveShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+
+sh:maxCount 4 ;    
+
+sh:datatype xsd:string ; 
+] .
+
+6.	Constraint for Transitivity 
+a.	Our constraint checks for transitivity in a graph.
+
+b.	An example of when one would use this constraint would be to check a graph of clan, band, nation affiliation. Such that when one is a recognized citizen of the Cherokee Nation one is also a member of one of the 3 bands of Cherokee and a member of one of the 7 clans. This constraint would be used to check if one is a member of a clan, they are a member of a band and if you are member a band you are a member of the Nation and if you are a member of a clan, you are a member of the Nation. One would simply change ex:TransitiveShape to ex:CherokeeNationEnrollmentClanTransitiveShape.
+
+c.	c -> b. b -> a. ؞ c -> a
+  i.	a = nation
+  ii.	b = band
+  iii.	c = clan
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:TransitiveShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 3 ;    
+
+sh:datatype xsd:string ; 
+
+] .
+
+7.	Constraint for Transitivity and Symmetry 
+a.	Our constraint checks for when a graph is both transitivity and symmetry in a graph. 
+b.	An example of when one would use this constraint would be to check a graph of clan, band, nation affiliation. Such that when one is a recognized citizen of the Cherokee Nation one is also a member of one of the 3 bands of Cherokee and a member of one of the 7 clans. This constraint would be used to check if one is a member of a clan they are a member of the a band and if you are a member of a band you are a member of the Nation and if you are a member of a clan you are a member of the Nation. Similarly, if you are a member of a clan you are a member of the Nation and if you are a member of the Nation you are a member of a clan.  
+c.	c -> b. b -> a. ؞ c -> a
+  i.	a = nation
+  ii.	b = band
+  iii.	c = clan
+d.	a -> c. c -> a
+  i.	a = nation
+  ii.	c = clan 
+
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:TransSymmetryShape
+
+a sh:NodeShape ;
```

Comment:
You can use the shapes you've already defined inside a complex shape like this, to save yourself work